### PR TITLE
refactor: create a new type to avoid repeat ourselves and add the correct return type for createClient method

### PR DESCRIPTION
**Pull request by ftonato**
## What kind of change does this PR introduce?

Refactor our codebase creating a new type to avoid repeat ourselves and add the correct return type for createClient method

### CODE REVIEW

**Comment by adam-beck** [comment]
Wouldn't `Record<string, string>` accomplish the same thing? And, since it's a built-in utility type, it would be easier to read and understand.

**Comment by ftonato** [comment]
> Wouldn't `Record<string, string>` accomplish the same thing? And, since it's a built-in utility type, it would be easier to read and understand.

You're totally right, they both do the same, it's just a matter of custom for those who have been using TS since before version 2.1 (which is when the records arrived) 🐶

About being easier to read, to me they both seem simple (again, seems like a particular preference)

File: src/SupabaseClient.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ export default class SupabaseClient {
   /**
    * Returns an array of all your subscriptions.
    */
-  getSubscriptions(): RealtimeSubscription[] {
+  _getSubscriptions(): RealtimeSubscription[] {
```
Comment (w3b6x9) [inline src/SupabaseClient.ts]: @ftonato i would keep it as `getSubscriptions` b/c developers can use it to grab their subscriptions and it'll break existing code (e.g. `client.getSubscriptions()`)

Comment (ftonato) [reply src/SupabaseClient.ts]: @w3b6x9 Sorry for the change! I thought the method was only used internally in the class, hence the idea of initialising it with underscore. Anyway, I "removed" the change and fixed the conflicts...

Comment (w3b6x9) [reply src/SupabaseClient.ts]: @ftonato no worries and thank you!

File: src/SupabaseClient.ts [hunk 2]
```diff
@@ -254,8 +255,8 @@ export default class SupabaseClient {
     })
   }
 
-  private _getAuthHeaders(): { [key: string]: string } {
-    const headers: { [key: string]: string } = this.headers
+  private _getAuthHeaders(): GenericObject {
+    const headers: GenericObject = DEFAULT_HEADERS
```
Comment (w3b6x9) [inline src/SupabaseClient.ts]: @ftonato i think the merge commit (https://github.com/supabase/supabase-js/pull/258/commits/19647931b45b6ee8a16c62a7608de826b18e91a6) messed this up:

```suggestion
    const headers: GenericObject = this.headers
```

File: src/lib/SupabaseQueryBuilder.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export class SupabaseQueryBuilder<T> extends PostgrestQueryBuilder<T> {
       table,
       fetch,
     }: {
-      headers?: { [key: string]: string }
+      headers?: GenericObject
```
Comment (w3b6x9) [inline src/lib/SupabaseQueryBuilder.ts:22]: @ftonato just one more thing. `GenericObject` needs to be imported.

**Review by w3b6x9 — CHANGES_REQUESTED** [review]

**Comment by ftonato** [comment]
@w3b6x9 First and last time I try to resolve conflict by browser hahaha

**Comment by w3b6x9** [comment]
> @w3b6x9 First and last time I try to resolve conflict by browser hahaha

haha wow mobile browser...that's quite the challenge!

**Review by w3b6x9 — APPROVED** [review]
lgtm! thanks!

**Pull request merged**